csvcubed/csvcubed/schema/cube-config-schema.jsonc

```diff
@@ -0,0 +1,303 @@
+{
+	"$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "http://gss-cogs.github.io/gss-cogs/csvcubed/csvcubed/schemas/cube-config-schema.jsonc",
+    "title": "Schema for a qb-config cube",
+    "description": "Schema to be used for creation & validation of a single csv-cubed-flavoured linked data cube in CSV-W format",
+    "type": "object",
+
+    "properties": {
+
+		"publisher": {
+            "description": "The schema publisher link uri",
+            "type": "string",
+            "format": "uri"
+		},
+
+        "creator": {
+            "description": "The schema publisher link uri",
+            "type": "string",
+            "format": "uri"
+        },
+
+        "spatial_bound": {
+            "description": "URI that defines the spatial / geographic bounds of the data contained herein",
+            "type": "string",
+            "format": "uri",
+            "default": "https://tdg.docbook.org/tdg/publishers/5.1/dcterms.html#dcterms.spatial"
+        },
+
+        "temporal_bound": {
+            "description": "URI that defines the temporal bounds of the data contained herein",
+            "type": "string",
+            "format": "uri",
+            "default": "https://tdg.docbook.org/tdg/publishers/5.1/dcterms.html#dcterms.temporal"
+        },
+
+        "columns": {
+            "description": "Columns in the same order as the csv, unspecified column types are assumed to be dimensions, case insensitive",
+			"properties": {
+				"Existing dimension": {"$ref": "#/$defs/existingDimension"},
```

Comment:
These are the example column names that @canwaf used in his JSON. The columns in CSVs won't necessarily have these names so we need to be able to support *any* column name mapping to any column type.

Reply:
There are some exceptions, though. We won't let anyone name a column "observations" and have it be a dimension. Additionally, there are some names we don't want (e.g. `code-list`, its variations, and other common RDF vocabulary used to construct URIs)

Reply:
> We won't let anyone name a column "observations" and have it be a dimension
I'm not sure we should be *that* prescriptive. If that's what they want to call a dimension, who are we to disagree?

Reply:
Okay. "Here's this rope, do what you will."